sql/hive/src/test/scala/org/apache/spark/sql/hive/InsertSuite.scala

```diff
@@ -728,4 +732,195 @@ class InsertSuite extends QueryTest with TestHiveSingleton with BeforeAndAfter
       assert(e.contains("mismatched input 'ROW'"))
     }
   }
+
+  test("[SPARK-21786] Check 'spark.sql.parquet.compression.codec' " +
```

Comment:
Could you split the whole test case to multiple independent smaller unit test cases?